src/geom/SpanSet.cc

```diff
@@ -147,24 +213,33 @@ SpanSet::SpanSet(std::vector<Span>&& vec, bool normalize) : _spanVector(std::mov
 
 void SpanSet::_runNormalize() {
     // This bit of code will only be executed with a non-empty _spanVector internally
-    // and is not accessable from outside the class
+    // and is not accessible from outside the class
 
     // Ensure the span set is sorted according to Span < operator
     std::sort(_spanVector.begin(), _spanVector.end());
 
-    // Now that the span is sorted, overlapping spans should be combined
-    // Start from 1 as the comparison will always be with the previous element
-    for (auto iter = ++(_spanVector.begin()); iter != _spanVector.end(); iter++) {
-        if (spansContiguous(*std::prev(iter, 1), *iter)) {
-            // These spans overlap, create one new one, covering the whole expanse
-            *(iter - 1) = Span((*(iter - 1)).getY(), std::min((*(iter - 1)).getMinX(), (*iter).getMinX()),
-                               std::max((*(iter - 1)).getMaxX(), (*iter).getMaxX()));
-            // Erase the current Span, as it is now contained in the previous element
-            iter = _spanVector.erase(iter);
-            // Move the iterator back one to make sure the element gets run in the loop
-            --iter;
+    // Create a new vector to hold the possibly combined Spans
+    std::vector<Span> newSpans;
+    // Reserve the size of the original as it is the maximum possible size
+    newSpans.reserve(_spanVector.size());
+    // push back the first element, as it is certain to be included
+    newSpans.push_back(*_spanVector.begin());
+
+    // With the sorted span array, spans that are contiguous with the end of the last span in the new vector
+    // should be combined with the last span in the new vector. Only when there is no longer continuity
+    // between spans should a new span be added.
+    for (auto iter = ++(_spanVector.begin()); iter != _spanVector.end(); ++iter) {
+        auto & newSpansEnd = newSpans.back();
+        if (spansContiguous(newSpansEnd, *iter)) {
+            newSpansEnd = Span(newSpansEnd.getY(), std::min(newSpansEnd.getMinX(), (*iter).getMinX()),
+                                 std::max(newSpansEnd.getMaxX(), (*iter).getMaxX()));
+        } else {
+            newSpans.push_back(Span(*iter));
         }
     }
+
+    // Set the new normalized vector of spans to the internal span vector container
+    _spanVector = newSpans;
```

Comment:
`std::vector::erase` is evil, or at least it is when it's used in a loop because you can get a huge number of moves. It's better to take this approach of creating a new vector containing the desired contents.